favourite-colour/options.js

```diff
@@ -6,8 +6,12 @@ async function saveOptions(e) {
 }
 
 async function restoreOptions() {
-  let res = await browser.storage.managed.get('colour');
-  document.querySelector("#managed-colour").innerText = res.colour;
+	try {
+		let res = await browser.storage.managed.get('colour');
+		document.querySelector("#managed-colour").innerText = res.colour;
+	} catch(error) {
+		console.log(JSON.stringify(error));
+	}
```

Comment:
@Rob--W something more like this?
```suggestion
  try {
    let res = await browser.storage.managed.get('colour');
    if (res.colour === undefined) {
      console.error("Data key colour not defined in the managed storage file.");
    }
    document.querySelector("#managed-colour").innerText = res.colour || "Data key colour not defined in the managed storage file.";
  } catch(error) {
    console.error("Managed storage file not found, check it's stored in an appropriate location.");
  }
```

Reply:
@rebloor, I took your suggestion and iterated on it a little.

 ```suggestion
  try {
    let res = await browser.storage.managed.get('colour');
    document.querySelector("#managed-colour").innerText = res.colour || "Could not find 'colour' in managed storage.";
  } catch(error) {
    if (error.message === "Managed storage manifest not found") {
      document.querySelector("#managed-colour").innerText = "Failed to find a managed storage manifest.";
    } else {
      document.querySelector("#managed-colour").innerText = `Unexpected managed storage error: ${error.message}`;
    }
  }
```